compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -583,6 +583,9 @@ impl<'tcx> InferCtxtBuilder<'tcx> {
 
     pub fn intercrate(mut self, intercrate: bool) -> Self {
         self.intercrate = intercrate;
+        if intercrate && self.tcx.use_next_trait_solver_in_coherence() {
+            self.use_next_trait_solver = true;
+        }
```

Comment:
vibe: can you instead add `fn use_next_trait_trait` to infcx builder and call that one when creatnig the inference context for coherence?

to keep any logic outside of the builder